MonoGame.Framework/Storage/StorageContainer.cs

```diff
@@ -118,9 +119,15 @@ internal StorageContainer(StorageDevice device, string name, PlayerIndex? player
 			// From the examples the root is based on MyDocuments folder
 #if WINDOWS_STOREAPP || WINDOWS_UAP
 			var saved = "";
-#elif LINUX || MONOMAC
+#elif MONOMAC
             // We already have a SaveData folder on Mac/Linux.
             var saved = StorageDevice.StorageRoot;
+#elif DESKTOPGL
+            string saved = "";
+            if(CurrentPlatform.OS == OS.Linux)
+                saved = StorageDevice.StorageRoot;
+            else

```

Comment:
How about we do something like this...
```
else if (CurrentPlatform.OS == OS.Windows)
{}
else
   throw new Exception("Unexpected platform!");
```
... or something so we get a good runtime error when the platform detection fails to work.

@KonajuGames @dellis1972  ?